lib/ibm/cloud/sdk_http/sdk_response.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: true
+
+require 'json'
+
+module IBM
+  module Cloud
+    # Module holds basic HTTP functionality.
+    module SDKHTTP
+      # Encapsulate the HTTP response.
+      # @param response [HTTP::Response] The HTTP response object.
+      class SDKResponse
+        def initialize(response)
+          @response = response
+        end
+
+        # The raw HTTP response.
+        # @return [HTTP::Response]
+        attr_reader :response
+
+        # Return the response in a hash or array.
+        # @return [Hash] When response is a hash.
+        # @return [Array] When response is an array.
+        # @raise [Exceptions::ExceptionWithResponse] Contents of body is not properly formatted json.
+        def json
+          JSON.parse(body, symbolize_names: true)
+        rescue StandardError
+          raise Exceptions::ExceptionWithResponse.new(self, "#{url} Error while parsing response body. #{response.body}")
+        end
+
+        # Return the raw response string.
+        # @return [String]
+        # @return [nil] Response does not have body method.
+        def body
+          response&.body.to_s
+        end
+
+        # Return the response code.
+        # @return [Integer] Response has code method.
+        # @return [nil] Response does not have code method.
+        def code
+          response&.code
+        end
+
+        alias status code
+
+        # Return the raw connection object.
+        # @return [HTTP::Connection]
+        # @return [nil] Response does not have a connection method.
+        def connection
+          response&.request
+        end
+
+        # Chainable method to verify the status code. Raise an exception for non 200-series or 404 status codes.
+        # @return [Response] Allows for method to be chainable.
+        # @raise [Exceptions::HttpStatusError] Raise if status checks failed.
+        def raise_for_status?
```

Comment:
As mentioned elsewhere, since this method doesn't return a boolean it probably shouldn't end in a `?`.  Since it raises, I think `!` is more appropriate.

Reply:
`validate_status!` might be a better name too, since in the chainable interface having the `raise` wording in the middle is a little strange, but if you use `!` that implies a raise might happen

Reply:
Fixed